homeassistant/components/media_player/kodi.py

```diff
@@ -130,6 +130,8 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         password=config.get(CONF_PASSWORD),
         turn_off_action=config.get(CONF_TURN_OFF_ACTION), websocket=websocket)
 
+    key = 'media_player.' + slugify(config.get(CONF_NAME))
+    hass.data[DATA_KODI][key] = entity
     async_add_devices([entity], update_before_add=True)
```

Comment:
I think it is not possible, since v0.40, the passed in `async_add_devices` method is a callback instead of a coroutine function. (https://home-assistant.io/blog/2017/03/11/repurpose-any-android-phone-as-ip-camera/)

I also did not like having to use `slugify` in that place, maybe it would be better to leave `hass.data[DATA_KODI]` as a list?